plugins/org.jkiss.dbeaver.ext.sqlite/src/org/jkiss/dbeaver/ext/sqlite/model/SQLiteUniqueKey.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.ext.sqlite.model;
+
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableBase;
+import org.jkiss.dbeaver.ext.generic.model.GenericTableConstraintColumn;
+import org.jkiss.dbeaver.ext.generic.model.GenericUniqueKey;
+import org.jkiss.dbeaver.model.runtime.VoidProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSEntityConstraintType;
+import org.jkiss.dbeaver.model.DBPInheritedObject;
+
+import java.util.List;
+
+public class SQLiteUniqueKey extends GenericUniqueKey implements DBPInheritedObject {
+
+    public SQLiteUniqueKey(GenericTableBase table, String name, @Nullable String remarks, DBSEntityConstraintType constraintType, boolean persisted) {
+        super(table, name, remarks, constraintType, persisted);
+    }
+
+    @Override
+    public boolean isInherited() {
```

Comment:
Wouldn't it make more sense to override `SQLTableManager#isIncludeIndexInDDL` and put this logic there? This feels hacky, indeed.

Reply:
Thanks for your review. The SQLTableManager#isIncludeIndexInDDL method cannot be overridden to correct this issue: in the SQLTableManager#getTableDDL the private method skipObject is used instead of isIncludeIndexInDDL to check if the primary key must be added or not...

Another solution could be to override SQLTableManager#getTableDDL to modify the DBPScriptObject.OPTION_SKIP_UNIQUE_KEYS option before generate the DLL, what do you think?

Reply:
You're referring to this part in the code, right?
https://github.com/dbeaver/dbeaver/blob/c40f4d5e4162e573f055ab3a72dd129517ae9034/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/edit/struct/SQLTableManager.java#L263-L275

You can add a new method similar to the one that checks an index for inclusion but for PKs. It's totally fine.

Reply:
> You're referring to this part in the code, right?
> 
> https://github.com/dbeaver/dbeaver/blob/c40f4d5e4162e573f055ab3a72dd129517ae9034/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/edit/struct/SQLTableManager.java#L263-L275

Yes

> You can add a new method similar to the one that checks an index for inclusion but for PKs. It's totally fine.

Ok, I'll try that, I wanted to avoid modifying generic classes to limit the impacts...

Reply:
> You can add a new method similar to the one that checks an index for inclusion but for PKs. It's totally fine.

That's what I did in the last commit...